User can view collection of products

miyeaier opened the pull request:
## PT Link: https://www.pivotaltracker.com/story/show/182009921

## User Story
```
As a visitor,
In order to get an overview of the products
I would like to be able to see a catalog
```
<img width="808" alt="Screenshot 2022-05-04 at 16 09 25" src="https://user-images.githubusercontent.com/54026570/166699845-e45a5492-b139-43c3-92a0-23aa9c1a2649.png">

<img width="1074" alt="Screenshot 2022-05-04 at 16 09 48" src="https://user-images.githubusercontent.com/54026570/166699873-24ae50d1-86e3-4117-b7f0-e9724b66b2a1.png">

<img width="763" alt="Screenshot 2022-05-04 at 11 52 57" src="https://user-images.githubusercontent.com/54026570/166660252-ed837ea9-fdcd-4538-a2d8-5969dac42d19.png">

```
PS: IF YOU CAN'T FIND "Create a menu component" Shagufta did on breach  header so everything is done for  User can view collection of products
```

miyeaier committed:
Create the cypress test file,Create a cypress  to stub out call to api,Add fixture  file of products,Make a  GET request to API,Map over return using  and display in jsx

miyeaier committed:
fix error message and got 3 list items and delete header component

miyeaier committed:
add image and description on products.jsx

tochman commented on .nyc_output/out.json:
```diff
@@ -94,6 +319,6 @@
     "b": {},
     "_coverageSchema": "1a1c01bbd47fc00a2c39e90264f33305004495a9",
-    "hash": "7c8e731bd4cc34f700a6cad6ff919c39f9cdbffb"
+    "hash": "deaffac37b29f0c57fa7e42cf2ce61997b0700d0"
```
This entire file needs to be removed from version control

miyeaier replied:
> This entire file needs to be removed from version control

removed now

tochman commented on cypress/fixtures/Products.json:
```diff
@@ -0,0 +1,27 @@
+      "description": " Red Nike Jersey Shirt "
+    }
+  ]
+}
```
Please use lower case letters in file names, unless it its a component

miyeaier replied:
changed

miyeaier replied:
> Please use lower case letters in file names, unless it its a component

changed

tochman commented on cypress/integration/UserCanViewProducts.feacture.js:
```diff
@@ -0,0 +1,10 @@
+});
+
+it("is expected to display a list with 3 items", () => {
+  cy.get("[data-cy=ProductList]").children().should("have.length", 3);
```
please use `snake_case` or `kebab-case` in data attributes. In thes case `product_list` or `product-list`

miyeaier replied:
> please use `snake_case` or `kebab-case` in data attributes. In thes case `product_list` or `product-list`

changed to kebab-case

tochman commented on cypress/integration/userCanSeeAnApplication.feature.js:
```diff
@@ -1,9 +1,12 @@
-  })
+    cy.intercept("GET", "**/api/products", {
+      fixture: "products.json",
+    }).as("getProducts");
```
There is no need to store the intercept in an variable. Please remove that.

tochman replied:
By the way, is this test still passing?

miyeaier replied:
yes

miyeaier replied:
> There is no need to store the intercept in an variable. Please remove that.

changed

tochman commented on src/App.jsx:
```diff
@@ -1,9 +1,12 @@
-    <div data-cy="name">Spoko</div>
-  )
-}
+    <div data-cy="products-list">
```
check your test file. This `cy` attribute is NOT what you are looking for in your test, or it rather, it is not used at all. Please see the `cy` attribute in `Products.jsx`

miyeaier replied:
> check your test file. This `cy` attribute is NOT what you are looking for in your test, or it rather, it is not used at all. Please see the `cy` attribute in `Products.jsx`

it works

tochman commented on src/components/Products.jsx:
```diff
@@ -0,0 +1,38 @@
+
+
+return (
+  <div data-cy="ProductList">{productList}</div>
```
See comment in test file regarding the `cy` attribute.

miyeaier replied:
> See comment in test file regarding the `cy` attribute.

no problem now

tochman reviewed (changes requested):
See comments. Also, I can't see that all tests are passing. Can't see that "Spoko"  is being displayed. But perhaps I'm missing something.

miyeaier committed:
change test file to the kebab-case and type eorro
add out.json to .git ignore
remove intercept in userCanSeeAnApplication.feature

miyeaier committed:
add .nyc_output/out.json on git ignore

miyeaier committed:
add .nyc_output/out.json to .gitignore file
change div to h1 on app.jsx

miyeaier committed:
add .nyc_output on gitignore

miyeaier commented:
> See comments. Also, I can't see that all tests are passing. Can't see that "Spoko" is being displayed. But perhaps I'm missing something.

miyeaier committed:
updete gitignore

tochman commented on cypress/integration/userCanViewProducts.feacture.js:
```diff
@@ -0,0 +1,20 @@
+beforeEach(() => {
+  cy.intercept("GET", "**/api/products", {
+    fixture: "products.json",
+  }).as("getProducts");
```
You are still saving the intercept but not using it in tests.

miyeaier replied:
fixed ,used now.

miyeaier committed:
add test get request to the API
